### ISSUE TITLE
Strange background color.

### ISSUE THREAD

**Issue by BurningSmile** [rakr/vim-one#60]
Hello,
I've been trying to figure out what is causing this issue when trying to use your color scheme. Applying the color scheme with set background=dark ends up looking like this.

https://i.imgur.com/jUCpuei.png

I have tried with 2 terminals, gnome-terminal and urxvt. both have 256 color support enabled.

I have tried creating a new vim install in a vm to verify it's not my personal configuration causing an issue.

output of vim --version
https://hastebin.burningsmile.tech/wosocugasi.diff

Thanks,

**Comment by OfficialOxide** [comment]
Try putting this in your .vimrc before `colorscheme one`:
`set termguicolors`.

**Comment by BurningSmile** [comment]
Ah, it needs to be before the set color scheme. Thanks

**Comment by OfficialOxide** [comment]
yeah, probably because of the tmux. try vim without tmux?

**Comment by BurningSmile** [comment]
Tested that last night, it results in this mess.
https://i.imgur.com/FicLxTc.png

**Comment by OfficialOxide** [comment]
Oh, wow.
Have you tried the same thing with other terminals? Though since you said you were using urxvt and gnome-terminal, it shouldn't happen.

can you post your `.vimrc`?

**Comment by BurningSmile** [comment]
Yeah, gnome-terminal works fine. Here is the ~/.vimrc.

https://hastebin.burningsmile.tech/apexefacoz

**Comment by OfficialOxide** [comment]
Try putting `syntax on` right after `call plug#end()`

**Comment by BurningSmile** [comment]
So it seems the black background color only apperars if the following conditions are met

1) rxvt-unicode-256 is used.
2) tmux is used in combination with eaither rxvt-unicode-256 or gnome-terminal.

gnome-terminal on its own is fine

**Comment by codyloyd** [comment]
I'm having the exact same issue here, with the black background on iTerm2.  it happens inside and outside of tmux.   I have played with many iTerm settings but none seem to actually resolve the issue :(

**Comment by krzkrzkrz** [comment]
Same result here with iTerm2

**Comment by rakr** [comment]
I think that iTerm2 supports true colours by default, so the only solution that I can see so far is to setup the colorscheme using the following sequence

```
set termguicolors     " enable true colors support
syntax enable
set background=light
colorscheme one
```

Make sure that your `$TERM` variable is at least `xterm-256color`

**Comment by MacgyverMartins** [comment]
@rakr thanks, this work for me

**Comment by rakr** [comment]
Keeping this issue open, but unfortunately I don't use a linux based computer. If someone in the linux community who uses Vim, gnome terminal and this theme can help that would be great.

**Comment by joshnroy** [comment]
@BurningSmile are you still having this issue with linux, gnome-terminal, tmux, and vim? I just fixed it and can provide help :)

**Comment by BurningSmile** [comment]
@joshnroy  Yup still having the issue since the last time i tested it, just tried with the newest copy of the theme. How did you go about fixing it?

**Comment by hamsternik** [comment]
I have the same issue as @BurningSmile  using iTerm2, Build 3.1.6 with Vim v.8.0 and without tmux. Just a black background.

My config [here](https://pastebin.com/iAwsA4YE).

**Comment by dgopsq** [comment]
Same issues with urxvt and neovim: black background with the true-color patch, lightning-blue otherwise.

**Comment by jmurinello** [comment]
Using `au ColorScheme one hi Normal ctermbg=None` seemed to have worked at first but then I realized it blasted the statusline colors when in insert or visual mode.

This sequence is what worked for me.
```vim
set termguicolors
syntax enable
let &t_8f = "\<Esc>[38;2;%lu;%lu;%lum"
let &t_8b = "\<Esc>[48;2;%lu;%lu;%lum"
set background=dark
colorscheme one
```
I have my `esc` mapped so that is why the color sequence starts with `\<Esc>` instead of `^[`.

I am using:
iTerm2 **Build 3.2.7**
Vim **v8.1.873**
Tmux **v2.8**

This is what the `$TERM` variable prints outside Tmux...
```sh
$ echo $TERM
xterm-256color
```
...and inside.
```sh
$ echo $TERM
screen-256color
```
Probably not relevant but this is the sequence for my `.tmux.config` 
```tmux
set -g default-terminal "screen-256color"
set-option -ga terminal-overrides ",xterm-256color:Tc"
```

Finally, I found [this](https://github.com/nathanbuchar/atom-one-dark-terminal/tree/master/scheme/iterm) repository with iTerm color presets for this colorscheme and  everything matches.

Hope it helps someone in the future.

**Comment by diegosomar** [comment]
@jmurinello confirmed, these steps work!

**Comment by Calvin-Xu** [comment]
I have the exact same issue on MacOS in iTerm2. Switching from the system Vim (/usr/bin/vim) to Homebrew Vim solves the issue. **Be aware that the system Vim does not have termguicolors**, whereas Homebrew Vim has it.

```
$ /usr/bin/vim --version
VIM - Vi IMproved 8.1 (2018 May 18, compiled Dec 13 2019 14:45:40)
Included patches: 1-503, 505-680, 682-1312
Compiled by root@apple.com
Normal version without GUI.  Features included (+) or not (-):
+acl               +extra_search      -mouse_netterm     -tag_old_static
-arabic            -farsi             +mouse_sgr         -tag_any_white
+autocmd           +file_in_path      -mouse_sysmouse    -tcl
+autochdir         +find_in_path      -mouse_urxvt       -termguicolors

[...]
```

**Comment by niknbr** [comment]
> Be aware that the system Vim does not support termguicolors

@Calvin-Xu Thank you for this. Been searching for hours and I was wondering why none of the popular solutions were working for me.
